packages/pluggableWidgets/datagrid-web/src/utils/__tests__/settings.spec.ts

```diff
@@ -225,6 +225,126 @@ describe("useSettings Hook", () => {
         rerender(initialProps);
         expect(props.settings.setValue).toHaveBeenCalledTimes(0);
     });
+
+    it("doesnt change the hooks when same properties are applied", () => {
+        const props = mockProperties();
+        const initialProps = {
+            settings: props.settings,
+            columns: props.columns,
+            columnOrder: ["0"],
+            setColumnOrder: props.setColumnOrder,
+            hiddenColumns: [],
+            setHiddenColumns: props.setHiddenColumns,
+            sortBy: [{ id: "0", desc: true }],
+            setSortBy: props.setSortBy,
+            filters: [{ id: "0", value: "ABC" }],
+            setFilters: props.setFilters,
+            widths: { "0": undefined } as ColumnWidth,
+            setWidths: props.setWidths
+        };
+
+        const { rerender } = renderHook(
+            ({
+                settings,
+                columns,
+                columnOrder,
+                setColumnOrder,
+                hiddenColumns,
+                setHiddenColumns,
+                sortBy,
+                setSortBy,
+                filters,
+                setFilters,
+                widths,
+                setWidths
+            }) =>
+                useSettings(
+                    settings,
+                    columns,
+                    columnOrder,
+                    setColumnOrder,
+                    hiddenColumns,
+                    setHiddenColumns,
+                    sortBy,
+                    setSortBy,
+                    filters,
+                    setFilters,
+                    widths,
+                    setWidths
+                ),
+            {
+                initialProps
+            }
+        );
+        // Initiates the hooks with values from settings once
+        expect(props.setColumnOrder).toHaveBeenCalledTimes(1);
+        expect(props.setHiddenColumns).toHaveBeenCalledTimes(1);
+        expect(props.setSortBy).toHaveBeenCalledTimes(1);
+        expect(props.setFilters).toHaveBeenCalledTimes(1);
+        expect(props.setWidths).toHaveBeenCalledTimes(1);
+        expect(props.settings.setValue).toHaveBeenCalledTimes(0);
+        rerender(initialProps);
+        expect(props.setColumnOrder).toHaveBeenCalledTimes(1);
```

Comment:
Initially it starter the loadedPreviousSettings with undefined, which causes the hooks to be called, right?

Reply:
Yes, that is correct. But after that you call the hook again with the same `initialProps`. The `previousLoadedSettings` is not set yet, so this code will evaluate to true, and, thus call the data grid hooks again:
https://github.com/mendix/widgets-resources/pull/338/files/f41285582521c02d5ed14c4d82f2113fbee9c72e..e5ce40439cb0967c9452b51aa1c0ce0fd56af3e6#diff-4b1e1ccf92252f3b25d8d486a301de301087aeb053be4cd81bc99754d641a193R71-R76

Reply:
No, this actually doesnt happens because the dependencies at line 102 do not receive updates, so this is executed once only. `[settings, filteredColumns, previousLoadedSettings.current]`

Reply:
Ah, yes that's right!